src/compile/scale.ts

```diff
@@ -41,35 +41,132 @@ export type ScaleComponents = {
   binColorLegend?: ScaleComponent
 }
 
-export function initScale(topLevelSize: number, mark: Mark, channel: Channel, fieldDef: ChannelDefWithScale, scaleConfig: ScaleConfig): Scale {
-  let scale: Scale = duplicate((fieldDef || {}).scale || {});
+export function channelScalePropertyIncompatability(channel: Channel, propName: string): string {
```

Comment:
I don't think my code perfectly matches the example in that case.  

There, toInt should return number.  My function returns a string describe incompatibility.  If there is no incompability, then undefined seems fine.  

Of course, you can rewrite this method in a similar fashion to that example, but TBH it seems like an overkill.  I'm not convinced that it's necessary better and worth my effort doing it.